vms/platformvm/txs/fee/calculator_test.go

```diff
@@ -0,0 +1,816 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package fee
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/snow"
+	"github.com/ava-labs/avalanchego/snow/snowtest"
+	"github.com/ava-labs/avalanchego/utils/constants"
+	"github.com/ava-labs/avalanchego/utils/crypto/bls"
+	"github.com/ava-labs/avalanchego/utils/crypto/secp256k1"
+	"github.com/ava-labs/avalanchego/utils/timer/mockable"
+	"github.com/ava-labs/avalanchego/utils/units"
+	"github.com/ava-labs/avalanchego/vms/components/avax"
+	"github.com/ava-labs/avalanchego/vms/platformvm/reward"
+	"github.com/ava-labs/avalanchego/vms/platformvm/signer"
+	"github.com/ava-labs/avalanchego/vms/platformvm/stakeable"
+	"github.com/ava-labs/avalanchego/vms/platformvm/txs"
+	"github.com/ava-labs/avalanchego/vms/platformvm/upgrade"
+	"github.com/ava-labs/avalanchego/vms/secp256k1fx"
+)
+
+var (
+	feeTestsDefaultCfg = StaticConfig{
+		TxFee:                         1 * units.Avax,
+		CreateAssetTxFee:              2 * units.Avax,
+		CreateSubnetTxFee:             3 * units.Avax,
+		TransformSubnetTxFee:          4 * units.Avax,
+		CreateBlockchainTxFee:         5 * units.Avax,
+		AddPrimaryNetworkValidatorFee: 6 * units.Avax,
+		AddPrimaryNetworkDelegatorFee: 7 * units.Avax,
+		AddSubnetValidatorFee:         8 * units.Avax,
+		AddSubnetDelegatorFee:         9 * units.Avax,
+	}
+
+	preFundedKeys             = secp256k1.TestKeys()
+	feeTestSigners            = [][]*secp256k1.PrivateKey{preFundedKeys}
+	feeTestDefaultStakeWeight = uint64(2024)
+	durangoTime               = time.Time{} // assume durango is active in these tests
+)
+
+type feeTests struct {
+	description         string
+	cfgAndChainTimeF    func() (StaticConfig, upgrade.Times, time.Time)
+	unsignedAndSignedTx func(t *testing.T) (txs.UnsignedTx, *txs.Tx)
+	expectedError       error
+	checksF             func(*testing.T, *Calculator)
+}
+
+func TestTxFees(t *testing.T) {
+	r := require.New(t)
+
+	tests := []feeTests{
+		{
+			description: "AddValidatorTx pre EUpgrade",
+			cfgAndChainTimeF: func() (StaticConfig, upgrade.Times, time.Time) {
+				eUpgradeTime := time.Now().Truncate(time.Second)
```

Comment:
I dislike the usage of `time.Now()` for unit tests since it's not the same across runs. I recommend just hard-coding a timestamp (say durango can be `time.Time{}` and the E upgrade can be `time.Time{} + 1 second`.

Reply:
So I defined a single upgrades.Config struct for all test cases, with hard-coded times. Each test can just set the chain time to select the upgrade it wants to test